Fix broken FX tests

datumbox opened the pull request:
Resolves [breakages](https://github.com/pytorch/pytorch/runs/7762125339?check_suite_focus=true) observed at #82560


Context:
The current FX tests assume that every public method under `torchvision.models` is a model builder method. To get a list of those methods, they query the `__dict__` attribute of the module. Unfortunately this assumption is not true and the tests already contain some workarounds to filter some methods. A better approach would be to query TorchVision for all of its available models under a specific module. This is exactly what the new Registration API can help us do and that's what we use in this PR.

datumbox commented:
So the tests [fail](https://github.com/pytorch/pytorch/runs/7772693290?check_suite_focus=true) with:
```
Traceback (most recent call last):
  File "test_fx.py", line 4206, in <module>
    TestVisionTracing.generate_tests()
  File "test_fx.py", line 4200, in generate_tests
    cls.generate_classification_tests()
  File "test_fx.py", line 4164, in generate_classification_tests
    for k in torchvision_models.list_models(module=torchvision_models):
AttributeError: module 'torchvision.models' has no attribute 'list_models'
Traceback (most recent call last):
  File "test/run_test.py", line 974, in <module>
    main()
  File "test/run_test.py", line 952, in main
    raise RuntimeError(err_message)
RuntimeError: test_fx failed!
```

But that's because we install an old version of TorchVision that doesn't contain the necessary `list_models()` method:
```
Successfully installed torchvision-0.14.0a0+1a1d509
```

The specific hashcode is the one last updated as seen at https://github.com/pytorch/pytorch/pull/82560. So basically we are in the situation where the test now fails because we can't install the TorchVision version that actually contains the new functionality for solving the issue. How do you fix this normally?

ezyang commented:
Update the torchvision hash as part of this PR. .github/ci_commit_pins/vision.txt

datumbox commented:
@ezyang Yeap that's what I did. Takes ages to export diffs today. Thanks for the tip.

datumbox committed:
Fix broken FX tests (#83187)

Summary:
Resolves [breakages](https://github.com/pytorch/pytorch/runs/7762125339?check_suite_focus=true) observed at https://github.com/pytorch/pytorch/issues/82560

Context:
The current FX tests assume that every public method under `torchvision.models` is a model builder method. To get a list of those methods, they query the `__dict__` attribute of the module. Unfortunately this assumption is not true and the tests already contain some workarounds to filter some methods. A better approach would be to query TorchVision for all of its available models under a specific module. This is exactly what the new Registration API can help us do and that's what we use in this PR.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/83187

Test Plan: buck1 test mode/dev //caffe2/test:fx

Reviewed By: ezyang

Differential Revision: D38583882

Pulled By: datumbox

fbshipit-source-id: a2b98596928f95fba7e4f34d047e0602f0a6c1f9
